source/fem-dem/gls_vans.cc

```diff
@@ -62,15 +78,30 @@ GLSVANSSolver<dim>::setup_dofs()
       if (this->simulation_parameters.boundary_conditions.type[i_bc] ==
           BoundaryConditions::BoundaryType::periodic)
         {
+          periodic_direction = boundary_conditions.periodic_direction[i_bc];
           DoFTools::make_periodicity_constraints(
             void_fraction_dof_handler,
             boundary_conditions.id[i_bc],
             boundary_conditions.periodic_id[i_bc],
-            boundary_conditions.periodic_direction[i_bc],
+            periodic_direction,
             void_fraction_constraints);
+
+          // Get periodic offset if void fraction method is qcm
+          if (this->cfd_dem_simulation_parameters.void_fraction->mode ==
+              Parameters::VoidFractionMode::qcm)
```

Comment:
Why is this only needed for QCM, and not also the other void fraction schemes? Is it only because of the volume averaging sphere?

Reply:
Yes, kind of since it calculates the distance between the quadrature points on one side of the PBC and the particles at the other side of the PBC. Since PCM doesn't "care" about the particles in cell neighbors, no needs of the offset to correct the particle position. But I just realize that SPM also needs  modifications since it also use particle position mhm.